switches/http/http.cpp

```diff
@@ -134,65 +135,89 @@ namespace Aseba
-        
-        // listen for incoming HTTP requests
-        httpStream = connect("tcpin:port=" + http_port);
+        // listen for incoming HTTP and Aseba requests
+        if (! inHttpPort.empty())
+            inHttpStream = connect("tcpin:port=" + inHttpPort);
+        if (! inAsebaPort.empty())
+            inAsebaStream = connect("tcpin:port=" + inAsebaPort);
+
+        // connect to each Aseba target
+        for (strings::const_iterator it = targets.begin(); it != targets.end(); it++)
+        {
+            try {
+                if (verbose)
+                    std::cout << "HttpInterface connect asebaTarget " << *it << "\n";
+                if (Dashel::Stream *cxn = connect(*it)) // triggers connectionCreated
+                {
+                    asebaStreams[cxn].clear(); // no node id until description is received
+                    Dashel::ParameterSet parser;
+                    parser.add("dummy:remapLocal=0;remapTarget=0;remapAesl=0");
+                    parser.add((*it).c_str());
+                    const unsigned localId(parser.get<unsigned>("remapLocal"));
+                    const unsigned targetId(parser.get<unsigned>("remapTarget"));
+                    const unsigned aeslId(parser.get<unsigned>("remapAesl"));
+                    // remember localId and aeslId as wishes to be answered when node description arrives
+                    if (localId > 0 && targetId > 0)
+                        localIdWishes[cxn][targetId] = localId;
+                    if (aeslId > 0)
+                        aeslIdWishes[cxn][targetId] = aeslId;
+                }
+            }
+            catch(Dashel::DashelException e)
+            {
+                std::cout << "HttpInterface can't connect target " << *it << ": " << e.what() << std::endl;
+            }
+        }
+      
+        // wait for descriptions
+        for (int i = 0; i < 20; i++) // 20 seconds
+        {
+            // ask for descriptions
+            this->pingNetwork();
+            this->run1s();
+            if (nodeDescriptionsReceived.size() >= targets.size())
+                break;
+        }
+
     }
     
     void HttpInterface::broadcastGetDescription()
     {
         GetDescription getDescription;
-        getDescription.serialize(asebaStream);
-        asebaStream->flush();
+        for (StreamNodeIdMap::iterator it = asebaStreams.begin(); it != asebaStreams.end(); it++ )
+        {
+            getDescription.serialize(it->first);
+            it->first->flush();
+        }
     }
     
-    bool HttpInterface::run2s()
-	{
-		int timeout(2000);
-		UnifiedTime startTime;
-		while (timeout > 0)
-		{
-			if (!step(timeout))
-				return false;
-			timeout -= (Aseba::UnifiedTime() - startTime).value;
-		}
-		return true;
-	}
-    
     void HttpInterface::run()
     {
         do
         {
             sendAvailableResponses();
-			pingNetwork();
-            if (!run2s())
-				break;
+            step(2);
```

Comment:
I admit that I missed adapting this from 31d63747853e5ed9c5c3e215102730c887d7afec. Testing with `if (!run1s()) break;` (line 275, but see my comment line 306)